### PR TITLE
fix #1328 - create table with DAO API

### LINKED ISSUE #1328: [API] DAO - create table in  SystemDB
When trying to create a table in ```SystemDB``` Dao-API creates it in ```DefaultDB```

### PR DESCRIPTION
**Pull request by iwolkov**
This PR allows creating tables with DAO API in local/SystemDB
Resolves #1328

### CODE REVIEW

File: api/api-javascript/api-database/src/main/resources/META-INF/dirigible/db/v4/query.js [hunk 1]
```diff
@@ -19,17 +19,9 @@ exports.execute = function(sql, parameters, databaseType, datasourceName) {
...
-		if (databaseType) {
-			if (datasourceName) {
-				resultset = org.eclipse.dirigible.api.v3.db.DatabaseFacade.query(sql, params, databaseType, datasourceName);
-			} else {
-				resultset = org.eclipse.dirigible.api.v3.db.DatabaseFacade.query(sql, params, databaseType);
-			}
-		} else {
-			resultset = org.eclipse.dirigible.api.v3.db.DatabaseFacade.query(sql, params);
-		}
+		resultset = org.eclipse.dirigible.api.v3.db.DatabaseFacade.query(sql,params,databaseType,datasourceName);
```
Comment (ThuF) [inline api/api-javascript/api-database/src/main/resources/META-INF/dirigible/db/v4/query.js:22]: Please double check what would happen, if the API is used in the following way:

```javascript
query.execute("SELECT * FROM DUMMY");
```

and also

```javascript
query.execute("SELECT * FROM DUMMY WHERE NAME = ?", ["Test123"]);
```

Then the connection should be to the default configured datasource connection (e.g. local/DefaultDB when no environment variables are provided)

File: api/api-javascript/api-database/src/main/resources/META-INF/dirigible/db/v4/update.js [hunk 2]
```diff
@@ -19,17 +19,9 @@ exports.execute = function(sql, parameters, databaseType, datasourceName) {
...
-		if (databaseType) {
-			if (datasourceName) {
-				result = org.eclipse.dirigible.api.v3.db.DatabaseFacade.update(sql, params, databaseType, datasourceName);
-			} else {
-				result = org.eclipse.dirigible.api.v3.db.DatabaseFacade.update(sql, params, databaseType);
-			}
-		} else {
-			result = org.eclipse.dirigible.api.v3.db.DatabaseFacade.update(sql, params);
-		}
+		result = org.eclipse.dirigible.api.v3.db.DatabaseFacade.update(sql,params,databaseType,datasourceName)
```
Comment (ThuF) [inline api/api-javascript/api-database/src/main/resources/META-INF/dirigible/db/v4/update.js:22]: Same as above

**Review by ThuF — CHANGES_REQUESTED** [review]
@iwolkov please double check the API behaviour, before merging the PR. Please also rebase it on the master branch, to solve the failing test issue.

File: api/api-javascript/api-database/src/main/resources/META-INF/dirigible/db/v4/query.js (hunk 1 above)
Comment (iwolkov) [reply api/api-javascript/api-database/src/main/resources/META-INF/dirigible/db/v4/query.js:22]: Checked. Works as expected.

File: api/api-javascript/api-database/src/main/resources/META-INF/dirigible/db/v4/update.js (hunk 2 above)
Comment (iwolkov) [reply api/api-javascript/api-database/src/main/resources/META-INF/dirigible/db/v4/update.js:22]: Checked. Works as expected.

**Review by ThuF — APPROVED** [review]
+1

**Pull request merged**